client/lib/memoize-last/index.ts

```diff
@@ -6,23 +11,35 @@
  *
  * @returns The wrapped function.
  */
-export default function memoizeLast< T extends ( ...args: any[] ) => any >( fn: T ): T {
-	let lastArgs: Parameters< T >;
-	let lastResult: ReturnType< T >;
+export default function memoizeLast< T extends ( ...args: any[] ) => any >( fn: T ): T & Clearable {
+	let lastArgs: Parameters< T > | symbol = UNSET_SYMBOL;
+	let lastResult: ReturnType< T > | symbol = UNSET_SYMBOL;
+
+	const func = ( ( ...args: Parameters< T > ) => {
+		if ( lastArgs === UNSET_SYMBOL ) {
```

Comment:
Unless I'm missing something, this code looks redundant. Wouldn't the existing code work as it was if you simply set `lastArgs` and `lastResult` to `undefined`, in `func.clear`? That would avoid the symbols and this `if` block altogether.

Reply:
my mistake, i thought args would be undefined if passed no args :)

Reply:
No worries, helped me learn an interesting technique! 😄